src/App.jsx

```diff
@@ -1,10 +1,53 @@
-import { useState } from 'react'
+import { useState, useEffect } from 'react';
 import './App.css'
+import MovieList from './MovieList'
+import Header from './Header'
+import Footer from './Footer'
 
 const App = () => {
-  <div className="App">
-    
+  const [loadedMovies, setLoadedMovies] = useState([]);
+  const [page, setPage] = useState(1); //Page number of next page to load
+  const baseURL = "https://image.tmdb.org/t/p/w500";
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const fetchMovies = async (prevMovieList, pages) => {
+    const apiKey = import.meta.env.VITE_API_KEY;
+
+    const response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?language=en-US&page=${pages}&api_key=${apiKey}`);
+    const data = await response.json();
+
+
+    setLoadedMovies(prevMovieList.concat(data.results));
+    // setLoadedMovies((prevMovieList) => [...prevMovieList, ...data.results] );
+  };
+
+  useEffect(() => {
+    fetchMovies(loadedMovies, page);
+  }, [page]);
+
+  const loadAdditionalPage = () => {
+    setPage(page + 1);
+    fetchMovies(loadedMovies, page+1);
+  }
```

Comment:
A few improvements could be made here:

1.  `useEffect` will re-run any time it's dependencies (`[page]`) change. This means that you don't need to call `fetchMovies` again after calling `setPage`, because the effect will take care of that for you.
2. Once the above change is made has been made, the function `fetchMovies` can be defined within the effect since it is only called there.
3. This one is more of a preference. The `loadAddtionalPage` function isn't reused and it is pretty short. So I personally would inline it directly in the JSX below. So like: `<MovieList movieList={loadedMovies} onClickLoadMore={() => setPage(page + 1)} />`